3.2 improvements to documentation (#68)
fsfe/reuse-website: site/content/en/faq.md

improve](https://fsfe.org/freesoftware/basics/4freedoms.html) the software, then
that software is [Free Software](https://fsfe.org/about/basics/freesoftware).

## What is a copyleft license? A permissive license? {#copyleft-permissive-license}

There are two broad categories of Free Software licenses; copyleft and
permissive licenses.

A copyleft license give you the right to use, study, share, and improve a piece
of software under the condition that, if you make any changes to the program,

> you must re-release those changes under the same license as the original. A

Author: I was aware writing this, but it's really difficult to write layman-approachable language here. I will give this another spin for accuracy's sake.

> you must re-release those changes under the same license as the original. A

Author: ```diff
 A copyleft license give you the right to use, study, share, and improve a piece
-of software under the condition that, if you make any changes to the program,
-you must re-release those changes under the same license as the original. A
-well-known copyleft license is the [GNU General Public License version
+of software under one strict condition. If you make any changes to the program
+and share your version of the program with someone else, you must share your
+changes under the same license as the original. A well-known copyleft license is
+the [GNU General Public License version
 3](https://www.gnu.org/licenses/gpl-3.0.html).
```

I think this'll do it. I cut the first sentence in half.

For the people for whom it matters that 'sharing'/'releasing' only extends to downstream; they're most likely companies that can afford lawyers. I don't see a pressing need to disambiguate in the REUSE FAQ.

> you must re-release those changes under the same license as the original. A

Reviewer: LGTM :)

license with an exception, you mark a file with the following tag:
`SPDX-License-Identifier: GPL-3.0-or-later WITH GCC-exception-3.1`.

> 

Reviewer: ```suggestion

You can find a list of common exceptions on [SPDX License Exceptions](https://spdx.org/licenses/exceptions-index.html).

```

## How does license compatibility work? {#license-compatibility}

> Free Software licenses are all different. Many licenses are compatible, meaning

Reviewer: ```suggestion
Free Software licenses often differ in detail. Many licenses are compatible, meaning
```

Many are more or less the same (e.g. think of all MIT/ISC/BSD/X11/…-likes).

And even if you compare e.g. AGPL and 0BSD, in the grand scheme of things they both offer the four freedoms, which is quite a huge and important part. The details can be quite important, but they remain details.

Additionally, many package hosting sites expect that you declare the licensing
information of your package. For instance, the [pyproject.toml file of the REUSE
tool](https://github.com/fsfe/reuse-tool/blob/main/pyproject.toml) declares
all the licenses that it uses in the format expected by the Python packaging
infrastructure.

## I only have a single license file. Should I still create a LICENSES directory? {#single-license}

Yes. This may seem extraneous, but it prevents future confusion when differently
licensed code is adopted. By keeping all licenses in a single directory, it is
easy for a user of your program to find all the licenses they need to comply
with in the blink of an eye.

> However, you can keep your `LICENSE`/`COPYING` file if you want. See [this

Reviewer: ```suggestion
However, in addition to that, you can keep your `LICENSE`/`COPYING` file if you want. See [this
```

comment header by hand, even though there is more information that should apply
to this file. A good example of this corner case is `.po` files provided by
translators. These may or may not contain copyright and licensing information,
depending on the translator's translation software and settings.

You can add additional information using the `precedence = "aggregate"`
key-value in your [`REUSE.toml` file](#bulk-license). An example looks like
this:

```toml
[[annotations]]
path = "po/*.po"
precedence = "aggregate"
SPDX-FileCopyrightText = "Contributors to My Project"
SPDX-License-Identifier = "GPL-3.0-or-later"
```

In the above example, 'Contributors to My Project' is added to the file's
copyright holders (if any), and 'GPL-3.0-or-later' is added to the file's
licenses (if any).

## ... copy someone else's work? {#copy-work}

If someone else has made their work available for you to use and copy, you can
incorporate their work into your project. When you put the work in one of the
files in your project, you should add an `SPDX-FileCopyrightText` tag for the copyright
holder(s) and an `SPDX-License-Identifier` tag for the license(s) under which

> the work was made available.

Reviewer: ```suggestion
the work was made available. Unless, of course, the author wishes to remain unattributed.
```

Author: This caveat doesn't really help the reader, I think. If we want to add advice on how to deal with anonymous copyright holders, it should be a separate question. Would you be willing to write that?

Reviewer: @carmenbianca , you mean a general section about posting _your own_ stuff pseudo-/anonymously or one that deals with also third party?

The issue I’d like to avoid _here_ is that people don’t think they need to go on a wild goose hunt to figure out who the author was of every little thing, if the authors themselves didn’t care signing their own work.

Author: I meant third-party stuff without names attached.

I think I understand better now why you want to add this, but:

- The proposed addition is a very roundabout way of saying that.
- I'm not sure it's a sound recommendation. Obviously, I don't hunt down every copyright holder either and usually make a best guess when it's especially arduous to figure out, but I don't feel at ease recommending that practice to other people, because I know it's not strictly The Best Thing To Do.

I'm convinced that there may be a way to instruct the user to not sweat it too much when the author didn't do their job of attributing themselves very well, but it needs a little more finesse.

Reviewer: There’s two use cases here I’d like to avoid:

- being more Catholic then the Pope – if the author didn’t care to make use of their rights, it’s not your obligation to correct that
- privacy breach – when the author very consciously did not want to include their name for whatever reason (e.g. _really_ does not care, safety reasons like dead-name or NDA issues)

Reviewer: That said, I am leaning towards agreeing with you not to complicate things in this section. I’ll suggest something in the following question/answer, as it may fit there better.

## ... exclude certain lines from REUSE compliance testing? {#exclude-lines}

In order to make the tool ignore a specific section containing strings that may
falsely be detected as copyright or license statements, you can wrap it within
the two comments `REUSE-IgnoreStart` and `REUSE-IgnoreEnd`.

<!-- REUSE-IgnoreStart -->

Please note that this must not be used to ignore valid copyright and licensing
information by yourself or a third party. The ignore blocks must only be used
for marking blocks that may trigger false-positive detections and errors.

An example for a file that contains commands or documentation that confuse the
REUSE tool:

```bash
# SPDX-FileCopyrightText: 2021 Jane Doe
#
# SPDX-License-Identifier: GPL-3.0-or-later

echo "SPDX-FileCopyrightText: $(date +'%Y') John Doe" > file.txt
echo "SPDX-License-Identifier: MIT" >> file.txt
```

This is how the section can be ignored:

```bash
# SPDX-FileCopyrightText: 2021 Jane Doe
#
# SPDX-License-Identifier: GPL-3.0-or-later

# REUSE-IgnoreStart
echo "SPDX-FileCopyrightText: $(date +'%Y') John Doe" > file.txt
echo "SPDX-License-Identifier: MIT" >> file.txt
# REUSE-IgnoreEnd
```

> <!-- REUSE-IgnoreStart -->

Reviewer: ```suggestion
<!-- REUSE-IgnoreEnd -->
```

…or am I missing something?

Author: this entire document is wrapped in REUSE ignore tags :) but the question on ignore tags keeps closing the document-wide ignore instruction, so we have to re-open it several times.

copyright holders (if any), and 'GPL-3.0-or-later' is added to the file's
licenses (if any).

## ... copy someone else's work? {#copy-work}

If someone else has made their work available for you to use and copy, you can
incorporate their work into your project. When you put the work in one of the
files in your project, you should add an `SPDX-FileCopyrightText` tag for the copyright
holder(s) and an `SPDX-License-Identifier` tag for the license(s) under which
the work was made available.

If the work was licensed differently from your project, you should [verify
whether the licenses are compatible](#license-compatibility), and add the new
licenses to your project.

## ... copy a work that has no copyright notice or license? {#no-copyright-license}

Before you proceed, always first make sure that you can find the copyright and
licensing information elsewhere. Some projects only include this information in
the root directory or in their README file.

If you can find no copyright notice, you can try to add it yourself by divining
the copyright holder from the author. See [this
question](#copyright-holder-author) for more information on the distinction.

> When in doubt, contact the author for clarification.

Reviewer: ```suggestion
When in doubt, contact the author for clarification.

It may also be that the author specifically wishes to remain pseudonymous or completely anonymous, which is their right. In this case, write their pseudonym, or if they do not want even that, then write `NONE`.
```

Another option would be `NOASSERTION`, but I’m leaning towards `NONE`.

From https://spdx.github.io/spdx-spec/v2.3/file-information/#88-copyright-text-field:

> Identify the copyright holder of the file, as well as any dates present. This shall be a free-form text field extracted from the actual file.
> 
> The options to populate this field are limited to:
> 
> Any text relating to a copyright notice, even if not complete;
> 
> NONE, if the file contains no copyright information whatsoever; or
> 
> NOASSERTION, if
> 
>     the SPDX document creator has made no attempt to determine this field; or
> 
>     the SPDX document creator has intentionally provided no information (no meaning should be implied from the absence of an assertion).
> 
> If the Copyright Text field is not present for a file, it implies an equivalent meaning to NOASSERTION. The metadata for the copyright text field is shown in Table 43.

Author: Looks good, added to the FAQ!

of that code, effectively making it impossible to, well, reuse that code. REUSE
helps you help other developers in reusing your code, which is one of the great
boons of Free Software.

## I'm used to doing things differently; can't I just keep doing what I have always been doing? {#tradition}

Some people, when they first encounter REUSE, run into a couple of things that
are very different from what they are used to. They miss the long legal blurbs
at the tops of files, miss the `COPYING`/`LICENSE` file in the root of the
repository, think that `.license` files clutter the directory, or find it very
strange that even insignificant non-code files get licensing headers. The
different-ness of REUSE can seem peculiar in that way.

One of REUSE's main goals is to be as compatible to existing best practices as
possible. It explicitly breaks with some old traditions, however. We believe
that REUSE is better for it. By breaking new ground, REUSE is simpler, more
comprehensive, and more unambiguous than it would have been if we tried to
maintain compatibility with traditions of old.

Nothing prevents you from adding long legal blurbs to your headers or putting a

> `COPYING` file in the root of your directory __anyway_ though. These things can

Reviewer: ```suggestion
`COPYING` file in the root of your directory _anyway_ though. These things can
```

be done to your liking. Some other aspects of REUSE are unavoidable, however.

> REUSE maintains that __every single file should contain licensing information_,

Reviewer: ```suggestion
REUSE maintains that _every single file should contain licensing information_,
```
